### PR TITLE
v2.8.0

### PR DESCRIPTION
**Pull request by HACO8888**
- Cover uploading

### CODE REVIEW

File: app/new/page.tsx [hunk 1]
```diff
@@ -121,6 +123,36 @@ export default function NewPostPage() {
...
   }
 
+  async function handleFileChange(e: ChangeEvent<HTMLInputElement>) {
+    const validImageTypes = ["image/webp", "image/jpeg", "image/png", "image/tiff"];
+
+    if (!e.target.files) return;
+
+    const file = e.target.files[0];
+    if (!validImageTypes.includes(file.type)) {
+      Swal.fire("File type not supported", "You can only upload  png,  jpg,  webp, tiff", "error");
```
Comment (banahaker) [inline app/new/page.tsx:133]: File type is not supported.

File: app/new/page.tsx [hunk 2]
```diff
@@ -121,6 +123,36 @@ export default function NewPostPage() {
...
+      return;
+    }
+
+    if (file.size > 5 * 1000 * 1000) {
+      Swal.fire("File too large", "You can only upload files under 5MB", "error");
```
Comment (banahaker) [inline app/new/page.tsx:138]: File size is too large.

File: app/new/page.tsx [hunk 3]
```diff
@@ -121,6 +123,36 @@ export default function NewPostPage() {
...
+      return;
+    }
+
+    try {
+      const res = await imageUpload(file, token);
```
Comment (banahaker) [inline app/new/page.tsx:143]:
```suggestion
      const imageUploadAPIResponse = await imageUpload(file, token);
```

File: app/new/page.tsx [hunk 4]
```diff
@@ -121,6 +123,36 @@ export default function NewPostPage() {
...
+      const res2 = await res.json();
```
Comment (banahaker) [inline app/new/page.tsx:144]:
```suggestion
      const imageUploadReponseJson = await res.json();
```

**Review by banahaker — CHANGES_REQUESTED** [review]

File: app/new/(new)/desktop/MetaDataForm.tsx [hunk 5]
```diff
@@ -84,9 +89,21 @@ export default function MetaDataForm({
               );
             })}
           </select>
-
+          <label className={styles.inputLabel}>Cover:</label>
+          <label htmlFor="file" className={styles.upload} style={{ cursor: (data?.cover) && "not-allowed" }}>
+            {(data?.cover !== "") ? "FIle Uploaded" : "No file uploaded"}
```
[the remark this answers is not in the document]
Comment (HACO8888) [reply app/new/(new)/desktop/MetaDataForm.tsx]: K

File: app/new/page.tsx (hunk 1 above)
Comment (HACO8888) [reply app/new/page.tsx:133]: K

File: app/new/page.tsx (hunk 2 above)
Comment (HACO8888) [reply app/new/page.tsx:138]: Not a problem

File: app/new/page.tsx (hunk 3 above)
Comment (HACO8888) [reply app/new/page.tsx]: Done

File: app/new/page.tsx (hunk 4 above)
Comment (HACO8888) [reply app/new/page.tsx]: Done

**Review by banahaker — APPROVED** [review]

**Pull request merged**